### ISSUE TITLE
Pipe operator

### ISSUE THREAD

**Issue by fritzblue** [vlang/v#6073]
[Too](https://docs.julialang.org/en/v1/base/base/#Base.:%7C%3E) [many](https://hexdocs.pm/elixir/Kernel.html#%7C%3E/2) [languages](https://github.com/tc39/proposal-pipeline-operator) realize the need for a pipe operator after already existing for a while. It would be amazing if V adopted this pattern earlier rather than later.

The [ECMAScript proposal](https://github.com/tc39/proposal-pipeline-operator#introduction) currently in flight provides a good introduction to how this feature operates and why it is so helpful. The short pitch is that it gives much greater readability when chaining multiple function calls together (rather than, for example, deeply nesting the function calls).

**Comment by JalonSolov** [comment]
That is an ugly operator, and completely contrary to everything else in V.  Not sure why is is called a "pipe operator" either, since it is used for chaining multiple functions.

V already supports "chaining" function calls, by simply doing `foo().bar().baz()`.  Example:
```v
println('hello, world'.capitalize().trim_right('ld')+'k!')
```
prints "Hello, work!"

**Comment by fritzblue** [comment]
What makes it "contrary to everything else in V"? Why would something so useful in the context of UNIX pipes, Haskell, F#, OCaml, Elixir, Elm, Julia, Hack, and others not be useful in V?

**Comment by JalonSolov** [comment]
The operator itself is unlike anything else in V.

Yes, Unix pipes are useful, but they will likely be done a different way.

**Comment by ntrel** [comment]
>  V already supports "chaining" function calls

No, it only supports method calls. I think this was closed without good reason, so I'm reopening it.

**Comment by ntrel** [comment]
@fritzblue it would help if you gave a motivating example in V of using this feature.

**Comment by spytheman** [comment]
thanks

**Comment by kulshekhar** [comment]
I would also like to voice support for this feature. My reason for supporting this is that it makes the code read a lot better.

The rescript (different language) docs for their pipe operator contains a good explanation of why this could be useful. https://rescript-lang.org/docs/manual/latest/pipe

> ```
> validateAge(getAge(parseData(person)))
> ```
> becomes
> ```
> person
>  ->parseData
>  ->getAge
>  ->validateAge
> ```

The reason I'm referring to rescript is because their implementation allows the piped functions to accept the argument at any location using a placeholder. So if you want the output of a function to be piped as the second argument of another function, that would be possible. For example:

```
input
  -> validateInput
  -> processInput('hard-coded value', _, 'another value')  
```
Here, the output of `validateInput` gets sent as the second argument to `processInput`

If this is implemented, it doesn't really matter what symbol is used for piping. It could be `->`, `|>` (my preference) or anything else. It could be whatever works best in the context of V

_Aside: If there are three things I could add in Go, it would be sum types, generics and pipe support. Out of those, V already has sum types and generics. Having pipe support would make it a no brainer (for me) to start using V once it stabilizes._

**Comment by jabcross** [comment]
I'd like to add that `|>` is adopted by quite a few languages for this purpose and looks pretty good if you have a ligature-enabled font like Fira Code.
